test/api/test_module_construct.cpp

```diff
@@ -1,16 +1,56 @@
+#include <numeric>
 #include <migraphx/migraphx.h>
 #include <migraphx/migraphx.hpp>
 #include "test.hpp"
 
+void run_test_on_gpu(const migraphx::program p)
+{
+    // make a copy of the program
+    migraphx_program_t p_ref_c;
+    migraphx_program_create(&p_ref_c);
+    migraphx_program_assign_to(p_ref_c, p.get_handle_ptr());
+    auto p_ref = migraphx::program(p_ref_c, migraphx::own{});
+    // compile on GPU
+    migraphx::compile_options options;
+    options.set_offload_copy();
+    p.compile(migraphx::target("gpu"), options);
```

Comment:
We dont need to run it on the gpu(and this will break when gpu is not available). Just run it on the ref and then check the expected result.

Reply:
Done.